airflow/providers/databricks/hooks/databricks_sql.py

```diff
@@ -241,13 +267,23 @@ def run(
         else:
             return results
 
-    @staticmethod
-    def _make_serializable(result):
-        """Transform the databricks Row objects into JSON-serializable lists."""
-        if isinstance(result, list):
-            return [list(row) for row in result]
-        elif isinstance(result, Row):
-            return list(result)
+    def _make_common_data_structure(
+        self, result: list[T] | T | None
+    ) -> list[T] | T | list[tuple] | tuple | None:
```

Comment:
As explained in the comment list[T] | T does not specify a common format. It just takes the input and returns the same. That does not make sense. `list[tuple] | tuple` should be sufficient. And I'd prefer it to be without `None` as that seems not compatible with a 'common_data_structure`

Reply:
As explained already in previous discussions I plan to add it in a follow up PR - together with ADR detailing some more of it. I think we should take a few more steps to clarify it and I even have issue defined here and assigned to me: https://github.com/apache/airflow/issues/36224

Perfect is an enemy of good. I suggest we complete this thing now and then add typing as seperate PR.

Is that ok @bolkedebruin  ? Can we do it this way and not circle back the same comment in this PR?

Reply:
I refactored `_make_common_data_structure` with the suggested typing, without None.